Templates/CppToolGem/Template/Code/Include/${Name}/${Name}TypeIds.h

```diff
@@ -0,0 +1,28 @@
+// {BEGIN_LICENSE}
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+// {END_LICENSE}
+
+#pragma once
+
+namespace ${SanitizedCppName}
+{
+    // System Component TypeIds
+    inline constexpr const char* ${SanitizedCppName}SystemComponentTypeId = "${SysCompClassId}";
+    inline constexpr const char* ${SanitizedCppName}EditorSystemComponentTypeId = "${EditorSysCompClassId}";
+
+    // Module derived classes TypeIds
+    inline constexpr const char* ${SanitizedCppName}ModuleInterfaceTypeId = "{${Random_Uuid}}";
+    inline constexpr const char* ${SanitizedCppName}ModuleTypeId = "${ModuleClassId}";
+    // The Editor Module by default is mutually exclusive with the Client Module
+    // so they use the Same TypeId
+    inline constexpr const char* ${SanitizedCppName}EditorModuleTypeId = ${SanitizedCppName}ModuleTypeId;
+
+    // Interface TypeIds
+    inline constexpr const char* ${SanitizedCppName}RequestsTypeId = "{${Random_Uuid}}";
```

Comment:
When the user adds another system component or anything with a TypeId, do they need to add it to this file if they want to use the new _DECL/_IMPL macros?  If so, it might be good to add a comment somewhere in here `Add the typeIds for your components and classes here`  Though I do miss the simplicity of having the type Id defined in the header with the DECL.

Reply:
Well the primary purposes of the TypeId header is to avoid the need to bring in header files with declarations of classes especially component classes.

The ComponentApplication uses a registration system that allows modules to register their components with it as a centralized location for component creation.
Any other module or system can then create these components without needing to know include any headers containing the definitions for the components.

This not only helps with compile times in other libraries, but also allows the module to better separate it's public API and private implementation.
Gems such as PhysX wouldn't need to expose it's PhysX Rigid Body Component as part of it's public API, but instead only needs to expose a TypeId. This allows greater flexibility for gem to perform actions such as change the C++ identifier name of component, refactor it's implementation, add any amount of includes within headers while only affecting the PhysX gem compile time and not downstream users of the gem, etc....

Also what should be noticed is that the `<GemName>TypeIds.h` contains no include files itself.
Making it cheap to include in other Gems as far as compile time is concerned.